### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by heltonmc**
General update to readme including accuracy estimates and benchmarks with lots more information.

Anything else we want to add or remove?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,59 +16,207 @@ pkg> add https://github.com/heltonmc/Bessels.jl.git
...
+julia> besselk0(x)
+1.6107849768886856e-6
+
+julia> besselk1(x)
+1.6750295538365835e-6
+```
+
+### Support for negative arguments
+
+Support is also provided for negative arguments and orders. However, if you use negative arguments, **real arguments could produce complex outputs**. See https://github.com/heltonmc/Bessels.jl/issues/30 for more details.
```
Comment (oscardssmith) [inline README.md:122]: I thought we didn't want to do this because it causes type stability issues?

Comment (heltonmc) [reply README.md:122]: Let me submit a PR and fix this to return error

Comment (heltonmc) [reply README.md:122]: Fixed in https://github.com/heltonmc/Bessels.jl/pull/32

Comment (heltonmc) [reply README.md]: And now updated with error message here.

**Comment by heltonmc** [comment]
Ok #32 is merged which makes everything type stable.

Are we ok with merging readme and registering?

**Comment by oscardssmith** [comment]
IMO, we should wait to register until we've transferred the repo, but I think this is good to merge.

**Comment by heltonmc** [comment]
Sounds good to me! I'm still uncertain how to do that. I tried transferring it to JuliaMath but I don't have write privileges in that org. Can you transfer it as a collaborator here?

**Comment by oscardssmith** [comment]
you would need to change my role from collaborator to an owner (this is different than transferring the repo)

**Comment by heltonmc** [comment]
Ahh thanks It looks like a private repo can only have one owner though. So would I need to transfer the repo to you first? Then you transfer to JuliaMath?

**Comment by oscardssmith** [comment]
that is surprisingly annoying. I think the double transfer might be the easiest.

**Comment by heltonmc** [comment]
Ya that is really annoying. I will transfer over to you then!

**Comment by heltonmc** [comment]
Ya I tried transferring but I got oscardssmith/Bessels.jl already exists.... come on GitHub lol

**Comment by oscardssmith** [comment]
My fork is now deleted. Try again?

**Comment by heltonmc** [comment]
Requested.

**Comment by oscardssmith** [comment]
transferred to JuliaMath.

**Comment by heltonmc** [comment]
Yay!!! Finally. Thank you! Is there a way to restore some of my contributor permissions for this?

**Comment by oscardssmith** [comment]
Yeah, just looking into that now. (It's somewhat complicated because I'm a collaborator, but not an owner of JuliaMath).

**Comment by heltonmc** [comment]
No problem!! Thanks for the help on this!

**Comment by oscardssmith** [comment]
should be fixed now.

**Comment by heltonmc** [comment]
Great! Yes looks like everything is good to go. Thank you!

**Comment by oscardssmith** [comment]
Register away!

**Comment by heltonmc** [comment]
https://github.com/JuliaRegistries/General/pull/65674

**Pull request merged**